op-service/txmgr/queue.go

```diff
@@ -0,0 +1,103 @@
+package txmgr
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/core/types"
+	"sync"
+)
+
+type TxReceipt[T any] struct {
+	Data    T
+	Receipt *types.Receipt
+	Err     error
+}
+
+type TxFactory[T any] func(ctx context.Context) (*TxCandidate, T, error)
+
+type Queue[T any] struct {
+	txMgr          TxManager
+	maxPending     uint64
+	pendingChanged func(uint64)
+	pending        uint64
+	cond           *sync.Cond
```

Comment:
I think that we could use a semaphore (`go/x/` [impl](https://pkg.go.dev/golang.org/x/sync/semaphore)) or a channel based version to simplify the internals here.

Reply:
finally thought i found a good use for `sync.Cond` 😄  but `Semaphore` makes much more sense... thanks